app/components/SchemaBrowserElements.tsx

```diff
@@ -0,0 +1,141 @@
+import { Link } from '@remix-run/react'
+import { Table } from '@trussworks/react-uswds'
+import { useState } from 'react'
+
+export type ReferencedSchema = {
+  $ref: string
+  type: string
+  schema?: Schema
+}
+
+export type SchemaProperty = {
+  description: string
+  type?: string
+  enum?: string[]
+  $ref?: string
+}
+
+export type Schema = {
+  $id?: string
+  $schema?: string
+  type: string
+  title?: string
+  description?: string
+  properties?: { [key: string]: SchemaProperty }
+  $defs?: { [key: string]: SchemaProperty }
+  anyOf?: ReferencedSchema[]
+  allOf?: ReferencedSchema[]
+  oneOf?: ReferencedSchema[]
+  required?: string[]
+}
+
+function ReferencedElementRow({ item }: { item: ReferencedSchema }) {
+  const [showHiddenRow, toggleHiddenRow] = useState(false)
+  const locallyDefined = item.$ref?.startsWith('#')
+
+  return (
+    <>
+      <tr onClick={() => toggleHiddenRow(!showHiddenRow)}>
+        {locallyDefined ? (
+          <>
+            <td>{item.$ref && item.$ref.split('/').slice(-1)[0]}</td>
+            <td>See below</td>
+          </>
+        ) : (
+          <>
+            <td>
+              <Link to={formatLinkString(item.$ref ?? '')}>
+                {item.$ref && item.$ref.split('/').slice(-1)[0]}
+              </Link>
+            </td>
+            <td>
+              {item.$ref && item.$ref.split('/').slice(-2)[0]} schema object{' '}
+              <small>(click to {showHiddenRow ? 'collapse' : 'expand'})</small>
+            </td>
+          </>
+        )}
+      </tr>
+      {!locallyDefined && item.schema && showHiddenRow && (
+        <tr>
+          <td colSpan={2}>
+            <Table fullWidth>
+              <thead>
+                <tr>
+                  <th>Name</th>
+                  <th>Type</th>
+                  <th>Description</th>
+                </tr>
+              </thead>
+              <SchemaPropertiesTableBody schema={item.schema} />
+            </Table>
+          </td>
```

Comment:
Rather than a nested table, can we have a single table with inherited fields indented?

Reply:
Probably, do you have any reference or something I could use as an example for what you are looking for?

Reply:
Sure, something like the macOS file browser. Notice how subfolders have their names indented, but the other columns are not.

<img width="1032" alt="Screenshot 2023-05-23 at 14 22 21" src="https://github.com/nasa-gcn/gcn.nasa.gov/assets/728407/1c02ef3d-e897-466e-8226-9e905a812ed5">

Reply:
What do you think of this?
![image](https://github.com/nasa-gcn/gcn.nasa.gov/assets/11023698/de0107d5-10e5-47b7-b0e9-7ba5b9195e2e)

Mobile:
![image](https://github.com/nasa-gcn/gcn.nasa.gov/assets/11023698/2c904713-184f-4548-ae79-a07b51a49f83)

Reply:
I like that. But add disclosure arrows to expand/collapse the sub tables.

Reply:
How is this?
![image](https://github.com/nasa-gcn/gcn.nasa.gov/assets/11023698/1cf1d0c6-0d64-4edb-9d13-ddd0f84ed995)